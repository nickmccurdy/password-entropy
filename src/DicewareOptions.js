import Options from './Options'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

export default class DicewareOptions extends PureComponent {
  static propTypes = {
    onChange: PropTypes.func.isRequired
  }

  possiblePasswords = () => 7776

  render () {
    return (
      <Options possiblePasswords={this.possiblePasswords} {...this.props}>
        <h3>Diceware</h3>
      </Options>
    )
  }
}
