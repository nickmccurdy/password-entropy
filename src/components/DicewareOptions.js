import Options from './Options'
import React, { Component } from 'react'

export default class DicewareOptions extends Component {
  static propTypes = Options.sharedPropTypes
  static shortName = 'Diceware'

  possiblePasswords = () => 7776

  render () {
    return (
      <Options possiblePasswords={this.possiblePasswords} {...this.props}>
        <h3>Diceware</h3>
      </Options>
    )
  }
}
