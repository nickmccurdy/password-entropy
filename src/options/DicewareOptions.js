import React from 'react'
import { FormControl } from 'react-bootstrap'

import FormGroup from '../ui/FormGroup'

export const dicewareWords = 7776

export default function DicewareOptions() {
  return (
    <FormGroup id="diceware" label="Possible Items" icon="question-circle">
      <FormControl.Static>{dicewareWords}</FormControl.Static>
    </FormGroup>
  )
}
