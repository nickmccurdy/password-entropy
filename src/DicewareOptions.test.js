import DicewareOptions from './DicewareOptions'
import { mount, shallow } from 'enzyme'
import React from 'react'

const callback = jest.fn()
const node = (<DicewareOptions onChange={callback}/>)

beforeEach(callback.mockReset)

it('renders', () => {
  expect(shallow(node)).toMatchSnapshot()
})

it('calls onChange prop', () => {
  mount(node)
  expect(callback).toHaveBeenCalledWith(7776)
})
