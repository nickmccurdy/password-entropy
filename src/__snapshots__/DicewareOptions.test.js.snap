// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Options
  onChange={[Function]}
  possiblePasswords={[Function]}
>
  <h3>
    Diceware
  </h3>
</Options>
`;
